contracts/Quest.sol

```diff
@@ -218,6 +249,14 @@ contract Quest is ReentrancyGuardUpgradeable, PausableUpgradeable, Ownable, IQue
         rewardToken.safeTransfer(sender_, amount_);
     }
 
+    /// @notice Internal function to update the referral reward amount
+    /// @param referrer_ The address of the referrer
+    function _updateReferralTokenAmount(address referrer_) internal {
+        uint256 referralAmount = this.referralRewardAmount();
```

Comment:
Let's push this into an internal function that gets called here and in the external function:
```suggestion
        uint256 referralAmount = _referralRewardAmount();
```

Then we do...
```
 function referralRewardAmount() external view returns (uint256) {
        return _refferalRewardMount();
    }
    function _referralRewardAmount() internal view returns (uint256) {
        return (referralRewardFee * rewardAmountInWei) / 10_000;
    }
```
    
This prevents us from "calling out" an external call triggered with `this.` will behave differently. Less of an issue when there's no calldata but still better form.

Reply:
Awesome thanks for this